crates/services/executor/src/executor.rs

```diff
@@ -440,13 +512,23 @@ where
             &mut data,
             &mut memory,
         )?;
-        self.process_l2_txs(
-            &mut partial_block,
-            &components,
-            &mut block_storage_tx,
-            &mut data,
-            &mut memory,
-        )?;
+
+        loop {
+            self.process_l2_txs(
+                &mut partial_block,
+                &components,
+                &mut block_storage_tx,
+                &mut data,
+                &mut memory,
+            )
+            .await?;
+            match self.new_tx_waiter.wait_for_new_transactions().await {
+                WaitNewTransactionsResult::Timeout => break,
+                WaitNewTransactionsResult::NewTransaction => {
+                    continue;
+                }
```

Comment:
Can we add some executor tests to show that this works?

Reply:
Agree, it would be nice to have it tested=)

Reply:
Added.